erigon-lib/trie/trie_test.go

```diff
@@ -635,3 +636,98 @@ func TestNextSubtreeHex(t *testing.T) {
 		assert.Equal(tc.expect, res, "%s, %s", tc.prev, tc.next)
 	}
 }
+
+func TestShortNode(t *testing.T) {
+	extensionKeyStr := "0b00000d020809020b080f0d0a090a0a0705070f050d0002090f0d0d0e07080e0402070e010c08030d0e0409060e040b0e0406060b0c080b0503010005"
+	extensionKeyNibbles := make([]byte, len(extensionKeyStr)/2+1)
+	for i := 0; i < len(extensionKeyStr)/2; i++ {
+		c, err := hex.DecodeString(extensionKeyStr[2*i : 2*i+2])
+		if err != nil {
+			t.Errorf("parsing error")
+		}
+		extensionKeyNibbles[i] = c[0]
+	}
+	extensionKeyNibbles[len(extensionKeyNibbles)-1] = 16
+
+	accRlpHex := "f84b08873f54314ab16fd0a056e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421a0c5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470"
+	accRlp, err := hex.DecodeString(accRlpHex)
+	if err != nil {
+		t.Fatalf("failed to decode rlp from hex")
+	}
+
+	valNode := ValueNode(accRlp)
+
+	shortNode := &ShortNode{Key: extensionKeyNibbles, Val: valNode}
+	trie := NewInMemoryTrieRLPEncoded(shortNode)
+	trie.valueNodesRLPEncoded = true
+	rootHash := trie.Root()
+
+	expectedHashStr := "fa7297cfa8b2d445e356e9f752e23b018ae8717fdf117cbe4fcef1b16217d3c2"
+	expectedHashBytes, err := hex.DecodeString(expectedHashStr)
+	if err != nil {
+		t.Fatalf("unable to decode expected hash %v", err)
+	}
+	if !bytes.Equal(rootHash, expectedHashBytes) {
+		t.Fatalf("rootHash(%x) != expectedHash(%x)", rootHash, expectedHashBytes)
+	}
+}
+
+//func TestIHCursorCanUseNextParent(t *testing.T) {
```

Comment:
those tests are not working? lets uncomment and add skip with reason or so

Reply:
Git blame shows @AskAlexSharov , I don't know why this is attributed to me.